lib/AsyncHttpClient/src/AsyncHttpClient.h

```diff
@@ -180,56 +184,59 @@ class AsyncHttpClient
      * @param[in] onError   Callback
      */
     void regOnError(const OnError& onError);
-    
+
     /**
      * Send GET request to host.
      *
+     * @param[in] userData Used to pass user-data which can be used in callbacks. 
      * @return If request is successful sent, it will return true otherwise false.
      */
-    bool GET();
+    bool GET(void* userData = nullptr);
 
     /**
      * Send POST request to host.
      *
+     * @param[in] userData  Used to pass user-data which can be used in callbacks. 
      * @param[in] payload   Payload, which must be kept alive until response is available!
      * @param[in] size      Payload size in byte
      *
      * @return If request is successful sent, it will return true otherwise false.
      */
-    bool POST(const uint8_t* payload = nullptr, size_t size = 0U);
+    bool POST(void* userData = nullptr, const uint8_t* payload = nullptr, size_t size = 0U);

```

Comment:
Optional userData will never be optional, because its on the most left side. In general move userData at the end (most right side) for all methods.